pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -93,7 +93,7 @@ func (r *resourceAllocationScorer) score(
 func calculateResourceAllocatableRequest(nodeInfo *schedulernodeinfo.NodeInfo, pod *v1.Pod, resource v1.ResourceName) (int64, int64) {
 	allocatable := nodeInfo.AllocatableResource()
 	requested := nodeInfo.RequestedResource()
-	podRequest := calculatePodResourceRequest(pod, resource)
+	podRequest := resourcehelper.GetResourceRequest(pod, resource)
```

Comment:
Let's first call `resourcehelper.GetResourceRequest` to get the **real** request. Based on that, since all resources have been aggregated properly, then call `GetNonzeroRequestForResource` to "adjust" the request - only for Scoring purposes.

Reply:
`resourcehelper.GetResourceRequest` returns the value of request directly. It isn't suitable for arguments of `GetNonzeroRequestForResource` function.

Reply:
true, you can tweak the method `GetNonzeroRequestForResource`, or make a new common method to fit the `podRequest` here.